buildtools/codeshift-js2ts.js

```diff
@@ -0,0 +1,991 @@
+                child_process
+                  .execSync(
+                    'npx -p typescript tsc /tmp/typedef.js --declaration --allowJs --emitDeclarationOnly --outDir /tmp/'
+                  )
+                  .toString()
+              );
+
+              result =
+                result.substring(0, path.value.start) +
+                fs.readFileSync('/tmp/typedef.d.ts') +
+                result.substring(path.value.end);
+            }
+          });
+      }
+      console.log('Convert the @enum by using the tsc command');
+      finish = false;
+      while (!finish) {
+        finish = true;
+        root = j(result);
+        root.find(j.VariableDeclaration).forEach((path) => {
+          if (finish) {
+            let comments = path.value.comments;
+            let hasExport = false;
+            if (
+              !comments &&
+              path.parent.value.comments &&
+              path.parent.value.type == 'ExportNamedDeclaration'
+            ) {
+              hasExport = true;
+              comments = path.parent.value.comments;
+            }
+
+            if (comments && comments[0].value.replace(/^[\n *]*/, '').startsWith('@enum')) {
+              fs.writeFileSync(
+                '/tmp/enum.js',
+                hasExport
+                  ? result.substring(comments[0].start, path.value.end)
+                  : `${result.substring(comments[0].start, comments[0].end)}\nexport ${result.substring(
+                      path.value.start,
+                      path.value.end
+                    )}`
+              );
+              console.log(
+                child_process
+                  .execSync(
+                    'npx -p typescript tsc /tmp/enum.js --declaration --allowJs --emitDeclarationOnly --outDir /tmp/'
+                  )
+                  .toString()
+              );
+
+              result =
+                result.substring(0, comments[0].start) +
+                fs.readFileSync('/tmp/enum.d.ts') +
+                result.substring(path.value.end);
+              finish = false;
+            }
+          }
+        });
+      }
+    }
+
+    console.log('Remove empty comments');
+    result = j(result)
+      .find(j.Comment)
+      .filter((path) => {
+        return path.value.type == 'CommentBlock' && path.value.value.replace(/[\n *]*/, '') == '';
+      })
+      .replaceWith((path) => {
+        return '';
+      })
+      .toSource();
+
+    error = false;
+    console.log('Format the result');
+    try {
+      const config = {
+        useTabs: false,
+        tabWidth: 2,
+        printWidth: 110,
+        singleQuote: true,
+        endOfLine: 'lf',
+        bracketSpacing: false,
+        quoteProps: 'preserve',
+        parser: 'babel',
+      };
+      return prettier.format(result, config);
+    } catch (error) {
+      console.log('Fail to format the result');
+      return result;
+    }
+    //  } catch (error) {
+    //    return result;
```

Comment:
Is it normal to catch error in the finally{} ?

Reply:
It's not in the finally, but I remove it...